build is failing on dotnet v3.0.100-preview4-011223 on azure devops

build is failing with the message -

```
...

Step 9/15 : RUN dotnet build --no-restore -c Release -o /app
 ---> Running in 275179430b90
Microsoft (R) Build Engine version 16.1.54-preview+gd004974104 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

/usr/share/dotnet/sdk/3.0.100-preview4-011223/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(151,5): message NETSDK1057: You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview [/src/src/Services/Identity/Identity.API/Identity.API.csproj]
Startup.cs(57,17): error CS1501: No overload for method 'UseRouting' takes 1 arguments [/src/src/Services/Identity/Identity.API/Identity.API.csproj]

Build FAILED.

Startup.cs(57,17): error CS1501: No overload for method 'UseRouting' takes 1 arguments [/src/src/Services/Identity/Identity.API/Identity.API.csproj]
    0 Warning(s)
    1 Error(s)
...

```

--- a/src/Services/Identity/Identity.API/Identity.API.csproj
+++ b/src/Services/Identity/Identity.API/Identity.API.csproj
@@ -1,12 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
-
   <PropertyGroup>
     <TargetFramework>netcoreapp3.0</TargetFramework>
   </PropertyGroup>
-
-
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.0.0-preview3-19153-02" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.0.0-preview4-19216-03" />
   </ItemGroup>
-
 </Project>

--- a/src/Services/Identity/Identity.API/Startup.cs
+++ b/src/Services/Identity/Identity.API/Startup.cs
@@ -33,7 +33,7 @@ namespace Identity.API
             });
 
 
-            services.AddMvc()
+            services.AddControllersWithViews()
                 .AddNewtonsoftJson();
         }
 
@@ -54,17 +54,18 @@ namespace Identity.API
             app.UseHttpsRedirection();
             app.UseStaticFiles();
 
-            app.UseRouting(routes =>
-            {
-                routes.MapControllerRoute(
-                    name: "default",
-                    template: "{controller=Home}/{action=Index}/{id?}");
-                routes.MapRazorPages();
-            });
-
             app.UseCookiePolicy();
 
+            app.UseRouting();
+
             app.UseAuthorization();
+
+            app.UseEndpoints(endpint =>
+            {
+                endpint.MapControllerRoute(
+                    name: "default",
+                    pattern: "{controller=Home}/{action=Index}/{id?}");
+            });
         }
     }
 }

